### ISSUE TITLE
Make support for `list`, `tuple`, `dict` ... type annotations

### ISSUE THREAD

**Issue by namannimmo10** [lcompilers/lpython#460]
Yes, we need to add `list` as a supported type annotation, so that we can create an empty list like this:
```console
(lp) namannimmo@srf lpython % cat t.py
def main():
  l: list
  l = []
  print(l)

main()

(lp) namannimmo@srf lpython % lpython t.py
semantic error: Unsupported type annotation: list
 --> t.py:2:3
  |
2 |   l: list
  |   ^^^^^^^ 

(lp) namannimmo@srf lpython % python t.py
[]
```

_Originally posted by @namannimmo10 in https://github.com/lcompilers/lpython/issues/455#issuecomment-1120365683_

**Comment by czgdp1807** [comment]
I have a couple of questions,

1. Do, we already support `list[f64]`, `list[i64]` at least in the ASR?
2. If the answer to above question is yes then what's the reason behind supporting just `list`. I think the intent of `list` is to allow a list with elements having not necessarily the same type. For example we will allow, `[1, 1.1, [1, 2], (0,)]`. However that means we want to allow dynamic typing because by just using `list` we are telling `lpython` that `l[0]` can be of any type (not known at compile time). So if we don't know the type at compile time, how can we even compile the code?

Let me know if I mis-interpreted anything. Thanks.

**Comment by certik** [comment]
1. I think so, or if not, we will.
2. Right now we only support one type for all elements. The list like `[1, 1.1, [1, 2], (0,)]` would be `list[any]`, with a dynamic, runtime type. We currently do not support that. We perhaps can, but that will be later. First we should focus on ahead of time compilation with static types.

**Comment by czgdp1807** [comment]
I agree. It will be better to focus just on static types for now and perform well in that direction. Once it reaches a good enough stage we can start thinking about dynamic typing as well. For now I will keep supporting `list[any]` or `list` lower than `list[i64]`, `list[f64]` in feature priority list.

**Comment by Smit-create** [comment]
@czgdp1807 This is done. We can close this.
